packages/block-editor/src/components/block-popover/index.js

```diff
@@ -75,6 +76,34 @@ function BlockPopover(
 		};
 	}, [ selectedElement ] );
 
+	// Get the scrollable container that the block popover appears within.
+	const scrollContainer = useMemo( () => {
+		if ( ! __unstableContentRef?.current ) {
+			return;
+		}
+		return getScrollContainer( __unstableContentRef?.current );
+	}, [ __unstableContentRef?.current ] );
+
+	// Force the block popover to re-render whenever the content area is scrolled.
+	// This ensures that the position of the popover is accurate for fixed or sticky blocks.
+	useLayoutEffect( () => {
+		if ( ! scrollContainer ) {
+			return;
+		}
+
+		scrollContainer?.addEventListener?.(
+			'scroll',
+			forceRecomputePopoverDimensions
+		);
+
+		return () => {
+			scrollContainer?.removeEventHandler?.(
+				'scroll',
+				forceRecomputePopoverDimensions
+			);
+		};
+	}, [ scrollContainer ] );
```

Comment:
Is this needed just for the toolbar? `BlockPopover` is used for a few different bits of UI.

`SelectedBlockPopover` is just for the selected block and handles the contextual toolbar, so it may be an optimisation to move the code there.

Though I don't think it has the `forceRecomputePopoverDimensions` function.

Reply:
It turns out I actually don't think we need any of the logic in the block popover component any more thanks to the changes that landed in https://github.com/WordPress/gutenberg/pull/46187 that resolved browser repaints of the popover when scrolling by switching transform values being passed to the animated wrapper. That's my hunch at least, but re-testing now against trunk, I can see this code isn't required now.

I'll re-do the description of this PR now that the scope of the change is a fair bit smaller (we just need to flipping logic now).